[network] 로그인 서버 연결 success 에 대해서 완료

minseungseon opened the pull request:
## 작업 내용 
- [ ] error handling 부분 외에는 완료함!

첫번째 서버 부분이라 고칠 점이나 잘못된 점 알려주시면 피드백 반영해서 수정해서 또 커밋하겠습니당 👍 

## 관계된 이슈, PR : 
x

## 스크린샷
x

## 레퍼런스
x

minseungseon committed:
[fix] DailyDiaryVC placeholder 문제 해결, 그 외 textView 관련 문제도 해결
body:
1. placeholder 가 다른데에 focus 갔다가 다시 돌아오면 Placeholder 가 사라지지 않는문제가 있어서 해결함
2. 직접 device test 를 했을 때 일기, 짧은 독서 뷰에서 글자를 쓴 후 화면 누르면 키보드 내려가는 기능이 없어서 추가함, @xwoud 의 코드 일부분 사용함.
3. [layout] 모든 뷰의 headerLabel 의 색깔이 지정 안되어 있었음. 설정해주었음.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into fix/placeholder

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
[style] 기상 인증 뷰 layout 완료
body:
dquote> 1. 이름과 같은 데이터전달이 될 매개변수는 기존에 @xwoud 님이 사용했던 변수 이름 그대로 userNick 으로 설정했음.
dquote> 2. uploadedPicture 는 UIImageView이며, 일단 test_img 파일을 assets 에 추가해서 연결해두었음. 실제 서버 연결에서는 회의 때 언급했던 kingfisher 의 image cashing 을 통해 이 부분에 데이터 전달을 하면
될 것으로 보임.
3. 22번째의 숫자도 매개변수화 해야할 것으로 보임.

minseungseon committed:
[refac] 숫자 변수화, letterSpacing 조정
body:
1. ~번째 의미있는 아침에서 숫자를 변수화 해주었음.
2. bodyUpperLabel 의 lineSpacing 을 조절하였음.. Zeplin 에 있는 것과 다른 숫자 개념인 것 같음. 일단 외관상 가장 비슷한 정도인 6 으로 line spacing 을 주었는데 이 부분에 대해서 이 extension 을 만든 본인이 검토를 해보도록 하겠습니다.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
[refac] 리뷰 피드백 반영
1. convention 수정
2. uploadedImage 데이터 전달 방식에 따라 변수화

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/myPage

minseungseon committed:
[style] 마이 피드 뷰 레이아
body:
1. rough 한 레이아웃만 잡았음
2. collection view 추가, nib 사용해보았음

minseungseon committed:
[refac] FeedImage model 추가, 연동 완료
body:
1. FeedImage struct 만들어두었음
2. @xwoud 의 피드백에 따라 ~번째 아침에서 숫자를 feedImageList 의 count로 설정하였음.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/myPage

minseungseon committed:
[fix] collection view의 scroll view inset 조정
dquote> body:
dquote> 1. iphone SE 외의 큰 device 에서 collection view 아래 부분 safe area 가 비어있게 스크롤이 되는 문제 해결
2. UIScrollViewContentInsetAdjustment .never 로 설정함
3. 간단한 주석 추가함

minseungseon committed:
[refac] collectino view 부분 코드 정리

minseungseon committed:
[refac] random 숫자0,1에 따라 profile 색깔 변하도록 적용함
body:
1. random number 적용'
2. subview 에서는 profile view 원으로 만들고, 색깔 적용
3. 아래 setLayout 에서는 textcolor 적용
+ subview 에서 textlabel 색깔을 한꺼번에 바꾸려고 하니까 적용이 안되었었음.. 왜그런지에 대해서 궁금함.
4. FeedCollectionViewCell.swift 주석처리 함

minseungseon committed:
[upload] 매일 오전 ~시 ~분 기상 UIView 를 위한 새로운 Extension 추가
body:
1. 이름의 새로운 extension file added.
2. 상세 설명 읽기 바람
3. 마이 피드 뷰에도 새로운 extension 적용 완료

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into refac/extension

minseungseon committed:
[refac] cell 에서 WakeUpTimeView Extension이 활용 불가능한 부분을 수정함.
body:
1. 매개변수를 dateData 하나만 받는 것으로 수정함
2. dateData 하나만 받고, 매일 오전 9시 기상 과 같은 string 을 반환하고, 반환되는 string 의 count 값으로 각자 적용하는 view controller 에서 view width 분기처리를 하기로 결정함.
3. UIViewController extension 이 toast 도 있기 때문에 이 둘을  라는 한 파일로 통합함.

minseungseon committed:
[style] 스플래시 lottie 추가 완료, LoginVC 연결
body: 1. lottie 파일은 스플래시용: 그리고 로그인 백그라운드 용:  이 업로드 됨.

minseungseon committed:
[style] 로그인 버튼 눌렀을 때 animation 추가함
body:
1. login 버튼을 처음 눌렀을 때에만 넘어가도록 해놓았

minseungseon committed:
[style] back button 눌렀을 때 다시 홈으로 animation을 통해 돌아가도록 설정함
body:
1. backbutton 을 누르면 animation 과 함께 모두 dissmiss

minseungseon committed:
[style] ID,PW Label, TextField 초기 높이 설정 조정
body:
초기 높이를 코드로 설정해주려고 했는데 계속 에러가 나서 스토리보드 오토레이아웃부터 조금 낮게 조정함

minseungseon committed:
[delete] reference 만 제거되어있었던 파일 제거

minseungseon committed:
[style] lint 위반하는 부분 수정, 작은 device 에 맞게 오토레이아웃 수정
body:
1. IDLabel 과 같은 것들 idLabel 로 수정
2. 작은 device 에서의 로그인, 비밀번호 란 위치 수정

minseungseon committed:
[style] login 뷰에서 비밀번호, 아이디가 틀릴 때 뜨는 알림 설정
body:
일단은 비어있을 때 알림이 뜨도록 설정해두었음

minseungseon committed:
[style] 로고없는 스플래시 배경 적용함, 로고이미지 띄우는 데에서 에러 발생
body:
LoginVC 에서와 동일하게 이미지를 적용시키려는데 에러가 발생함.

minseungseon committed:
[refac] 머지 충돌 해결 완료, 애니메이션 수정
body:
머지 충돌 해결, back button 누르고 다시 돌아왔을 때 textfield 설정에서 오타로 인한 오작동이 있는거 확인하고 수정함

minseungseon committed:
[refac] scene delegate 수정
body:
scene delegate 에서 root vc 주석 처리함

minseungseon committed:
[style] login 뷰 완료

minseungseon committed:
[style] login 뷰 완료

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into refac/PictureUpload

minseungseon committed:
[refac] 기상 인증 뷰 뷰디자인 수정에 따른 대응 처리
body:
1. 뷰에 들어가야하는 컴포넌트가 다소 많아서 SE 의 경우에는 '한 화면을 눌러서 수정하세요' 글씨를 넣지 않음
2. 개인적 욕심으로 textview 편집 시작하면 '흰 화면을 눌러서 수정하세요' 글씨 사라지게 처리해둠
3. iphone mini 에서도 아래가 조금 좁은 느낌이 들어서 이미지와 위 라벨(~꾸준히 기록해보아요) 사이 간격을 제플린 기준보다 5 만큼 줄였음

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into refac/PictureUpload

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 1st/loginServer

minseungseon committed:
[network] LoginData.swift 추가, API Service Extension 추가
body:
LoginVC 에서 extension 부분 추가

minseungseon committed:
[network] 간단한 로그인 성공에 대한 서버 연결 완료
body:
judgeObject 를 통해 데이터 처리를 해줌, 성공시에 홈뷰로 넘어가도록 일단 설정해두었음.

minseungseon committed:
[network] 로그인 서버 연결 success 에 대해서 완료
body:
error handling 추가해야함

minseungseon committed:
[upload] merge 커밋

xwoud commented on meaning/meaning/Screen/Login/ViewController/LoginVC.swift:
```diff
@@ -304,6 +316,69 @@ class LoginVC: UIViewController {
+                    return
+                }
+                
+                // userNick, wakeupTime 폰에 저장
+                if let userNick = loadData.userNick,
+                   let wakeupTime = loadData.wakeUpTime {
+                    UserDefaults.standard.setValue(userNick, forKey: "userNick")
+                    UserDefaults.standard.setValue(wakeupTime, forKey: "wakeUpTime")
+                    
```
두개의 token값은 저장하지 않나요?!

minseungseon replied:
오우 추가하겠습니다!!!

xwoud commented on meaning/meaning/Screen/Login/ViewController/LoginVC.swift:
```diff
@@ -11,17 +11,20 @@ import Lottie
         NSAttributedString.Key.kern : -0.48
     ]
+    var loginData: LoginData?
+    var userNick: String?
+    var wakeupTime: String?
```
요 두개는 만들어졌지만 사용되지 않는거같아요! 삭제해도 되지 않을까요?

minseungseon replied:
헉 그렇군... 넵!!!

xwoud commented on meaning/meaning/Screen/Login/ViewController/LoginVC.swift:
```diff
@@ -304,6 +316,69 @@ class LoginVC: UIViewController {
+
+                }
+                
+                self.loginData = loadData
```
불필요한 코드 아닐까요? 뷰컨에서 선언해둔 ```loadData```에 넣은 후 하는 것이 없어보입니다 !

minseungseon replied:
앗 넵!!!

xwoud reviewed (commented):
첫 moya를 이용한 서버통신! 어려웠을텐데 아주 잘해서 눈물 줄줄🥲

pk33n reviewed (approved):
저는 보고 배우겠습니다 민승 최고

minseungseon committed:
[network] 로그인 에러핸들링까지 완료
body:
1. UserDefaults 값도 저장이 잘 되는 것 확인함
2. 400 error 에 대해서는 뷰가 넘어가지 않고, 아이디 비밀번호를 다시 확인하도록 했음
3. 그 외 에러에 대해서는 '네트워크 확인' 에 대한 toast 알람 설정을 함

minseungseon committed:
[style] 스플래시 뷰 로고 추가
body:
그동안 로고가 안떴던 문제 해결, app delgate 에서 스토리보드를 설정안해준 잘못이었음.
또한 @xwoud 언급대로 뷰를 하나 띄우고 그 위에 보여주는 식으로 함.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 1st/loginServer

minseungseon committed:
[docs] 주석 정리
